Code/Tools/FBuild/FBuildCore/Helpers/VSProjectGenerator.cpp

```diff
@@ -86,6 +88,48 @@ void VSProjectGenerator::AddFiles( const Array< AString > & files )
     guid.Format( "{%08x-6c94-4f93-bc2a-7f5284b7d434}", CRC32::Calc( projectNameNormalized ) );
 }
 
+// IsFilenameAHeaderFile
+//------------------------------------------------------------------------------
+static inline const bool IsFilenameAHeaderFile( const AString & fileName )
+{
+    // TODO: Add support for other header types?
+    return ( fileName.EndsWith( 'h' ) || fileName.EndsWith( ".hpp" ) || fileName.EndsWith( ".hxx" ) || fileName.EndsWith( ".inl" ) );
+}
+
+static inline const bool ShouldWeSkipThisPlatform( const AString & platformIncludePattern, const AString & platform )
+{
+    // If we have a .PlatformIncludePattern filter defined, and this platform doesn't match it, return true [So we can skip emitting this platform to the .vcxproj]
+    const char *thisPlatformIncludePattern = platformIncludePattern.Get();
+    ASSERT( (thisPlatformIncludePattern != nullptr) && (platformIncludePattern != "") );
+    return ( (!platformIncludePattern.IsEmpty()) && (!platform.Find( thisPlatformIncludePattern )) );
+}
+
+static inline const char* GetPossiblyAliasedPlatformStringInUppercase( const AString & targetArchitectureAlias, const AString & platform )
+{
+    // Handle Platform Aliases - these define new platforms and alias existing platforms to them
+    if( !targetArchitectureAlias.IsEmpty() )
+    {
+        const char *thisPlatformAlias = targetArchitectureAlias.Get();
+        ASSERT( (thisPlatformAlias != nullptr) && (targetArchitectureAlias != "") );
+        AStackString<>							thisPlatformAliasString( thisPlatformAlias );
+                                                thisPlatformAliasString.ToUpper();
+        const char *thisPlatformAliasUpper =	thisPlatformAliasString.Get();
+        return thisPlatformAliasUpper;
```

Comment:
This is returning a pointer into the stack that falls out of scope when this function returns. This is definitely a bug.